Change redirect target to use the full URL with protocol and perform …

andrewchiles opened the pull request:
Thanks for this project! We found that we needed more control of our redirect target by specifying the protocol along with the destination (e.g., a bad user agent requests our site with https, but we want to redirect them to a plaintext http destination, which is different than the original `REQUEST_SCHEME`).

This is incompatible with the current version of the project since specifying a full URL to `-d` results in redirects like `https://http://example.com/index.html`, which obviously fail. 

We fixed this behavior by removing various references to `%{REQUEST_SCHEME}://` (including those within the `.htaccess` source  and verifying that the supplied redirect destination is a valid URL on startup. 

**Summary**
- Change  `-d` redirect target to use the full URL with protocol
- Perform URL validation on the `-d` parameter
- Standardize use of  `${REDIR_TARGET}` instead of `%{REQUEST_SCHEME}://${REDIR_TARGET}` so that we have more control over the target for redirection
- Modify sys.exit() calls to return error code `1` when an error condition was met
- Fix typo

andrewchiles committed:
Change redirect target to use the full URL with protocol and perform URL validation

0xZDH commented on redirect_rules.py:
```diff
@@ -571,6 +584,5 @@
-
     elapsed = time.perf_counter() - start
-    print(f"\n{__file__} executed in {elapsed:0.2f} seconds.")
+    print("\n{__file__} executed in {elapsed:0.2f} seconds.")
```
Was there a reason for removing the f-string? This change causes the final execution timer displayed to show the explicit string above instead of the file name and elapsed time.

andrewchiles replied:
Ah good catch, I'll add the `f` back

0xZDH commented on redirect_rules.py:
```diff
@@ -30,7 +30,7 @@
+    sys.exit(1)
 
 
 __version__ = '1.2.2'
```
Bump the version to 1.2.3.

andrewchiles replied:
Fixed

0xZDH commented on redirect_rules.py:
```diff
@@ -79,12 +79,21 @@
 
+# URL Regex
+regex = re.compile(
+        r'^(?:http|ftp)s?://' # http:// or https://
```
Include in the comment 'ftp:// or ftps://' for better awareness of what his regular expression is doing.

andrewchiles replied:
Fixed in subsequent commit to remove ftp/ftps altogether as that doesn't really fit this redirection scenario

0xZDH reviewed (changes requested):
This is a fantastic update - thanks for the PR! I added a few comments - please review and update and I will get this merged into the main branch.

andrewchiles committed:
Update redirect_rules.py

Address comments at: https://github.com/0xZDH/redirect.rules/pull/5#pullrequestreview-564914478
- Replace missing `f` on print statement
- Bump version to 1.2.3

andrewchiles committed:
Update redirect_rules.py

Remove `ftp` and `ftps` from allowed redirect targets

andrewchiles commented:
Thanks for taking time to review, all comments were addressed with the latest commits!

0xZDH reviewed (approved):
Looks good!
